code/mob/living/critter/sawfly.dm

```diff
@@ -177,7 +186,7 @@ This file is the critter itself, and all the custom procs it needs in order to f
 			explosion(src, get_turf(src), 0, 0.75, 1.5, 3)
 			qdel(src)
 
-		if(isalive(src)) // prevents weirdness from emagged controllers causing frankenstein sawflies
+		if(isalive(src)) // if they get EMP'd, they don't die, we we'll want to fix that
 			qdel(src)
```

Comment:
seems kind of weird for them just vanish 33% of the time when emp'd

Reply:
this is in the blowup proc, which is called by the EMP_act proc 2 seconds after they get EMP'd. The qdel is to ensure they clean upproperly when exploded after the EMP, since they don't fully die and having explode to leave a body behind is odd.